lib/orders.js

```diff
@@ -470,21 +470,63 @@ function captureOrder(args) {
  * @param {string} args.clientType - Client type to create order (WEBHOOK_CLIENT | POLLING_CLIENT) 
  * @param {string} args.customClientId - Custom client id to create order
  * @param {string} args.customClientSecret - Custom client secret to create order 
+ * @param {string}  args.scheme - Payment scheme selected (ideal, oxxo, etc.)
  * @returns {Promise} Promise object represents the client id and secret
 */
 function determineClientId(args) {
 	return new Promise((resolve, reject) => {
 
-		const { environment, clientType, customClientId, customClientSecret } = args;
+		const { environment, clientType, customClientId, customClientSecret, scheme } = args;
 
 	    const IS_WEBHOOK_CLIENT = clientType == 'WEBHOOK_CLIENT' ? true : false;
 
 	    const webHookClientTag = IS_WEBHOOK_CLIENT ? "PRIMARY" : "NO_WEBHOOK";
+
+		let clientId, clientSecret
 	    
 	    // Check if custom client id was entered; if not, use default from environmental configs
 	    if (!_.isEmpty(customClientId)) {
 	    	resolve({clientId: customClientId, clientSecret: customClientSecret });
-	    } else {
```

Comment:
For the different schemes, is there any scheme specific logic here other than getting the scheme specific variable from the config file? Can we get rid of the switch statement and allow any scheme client id to be overridden by the config file?

```suggestion
}  else  {
   clientId = process.env[`PP_${environment}_CLIENT_ID_${scheme.toUpperCase()}`] || process.env[`PP_${environment}_CLIENT_ID_${webHookClientTag}`];
   clientSecret = process.env[`PP_${environment}_CLIENT_SECRET_${scheme.toUpperCase()}`] || process.env[`PP_${environment}_CLIENT_SECRET_${webHookClientTag}`];
   resolve({ clientId, clientSecret });
}
```
If the scheme specific variable doesn't exist, it will just default to the environment default for primary or webhook.

This change would mean that any env override would need to match the spelling of the scheme though. So `PP_LIVE_CLIENT_ID_BOLETO` would need to change to `PP_LIVE_CLIENT_ID_BOLETOBANCARIO`

Reply:
Apply this for Live environment only?

Reply:
changed to apply for all apms and environments